sdk/program/src/stake/tools.rs

```diff
@@ -0,0 +1,34 @@
+//! Utility functions
+
+/// Helper function for programs to call [`GetMinimumDelegation`] and then fetch the return data
+///
+/// This fn handles performing the CPI to call the [`GetMinimumDelegation`] function, and then
+/// calls [`get_return_data()`] to fetch the return data.
+///
+/// [`GetMinimumDelegation`]: super::instruction::StakeInstruction::GetMinimumDelegation
+/// [`get_return_data()`]: crate::program::get_return_data
+pub fn get_minimum_delegation() -> u64 {
```

Comment:
Looks like `invoke()` in non-bpf always returns `Ok`, so I've just put a cfg block around the unwrap for `get_return_data()`. How's that?

Reply:
I think I led you astray with suggesting the `unwrap`, I think returning a `ProgramResult` is better because programs are used to handling those anyways. This is the approach taken by the ATA getter method as well: https://github.com/solana-labs/solana-program-library/blob/1d1c2b178b8cf2ed3e28006c27b2ba5b3d039d67/associated-token-account/program/src/tools/account.rs#L76

My suggestion here of adding a cfg attribute to the function wasn't good because if developers use the program test framework in non-bpf environments, their programs wouldn't compile if they used this function.

Your change looks ok but I'm personally not a fan of returning 0 if we fail to get the return value. I think an error is better. So might as well do away with the cfg stuff and just return a result here I think.

Reply:
Result or Option though? Invoke can fail (Result), and get return data may not match, so I felt Option was a better fit. I could return Result<Option<u64>>, but that feels convoluted. Whatcha think?

Reply:
Result is probably easiest for programs to handle, just need to add a `?`

Reply:
Agreed. `ProgramResult` is the Right Thing :tm: to return

Reply:
What `ProgramError` should I return if `get_minimum_delegation_return_data()` fails?

Reply:
And to be clear, `ProgramError` _not_ `ProgramResult`, since `ProgramResult` returns `()` for `Ok` (and I explicitly want `u64`).

Reply:
Sorry yes, you're right we don't want `ProgramResult`. For choosing which `ProgramError` to us, I would mimic what the ATA helper does

Reply:
Done!